### DISCUSSION TITLE
TFTP timeout on fresh Lubuntu 16.04 LTS with LTSP

### DISCUSSION THREAD

**Discussion by costinpc** [ltsp/ltsp#481] [Q&A] [answered]
Hello everyone!
First I have to say I'm a newbie to linux but I try to learn.
I have the following scenario: One server (i7-2600, 8Gb RAM, 2 NIC-s) and 20 clients (Pentium Dual Cores ~2Ghz, 1 GB RAM). 
Clients need only browsing. I installed Lubuntu 16.04 LTS and LTSP following install instructions found at ltsp.org/docs/installation/. 
Everything seems to be ok /etc/dnsmasq.d/ltsp-dnsmasq.conf but when clients try to boot they get TFTP timeout error, and strange thing is that if i restart dnsmasq.service and try to boot again it works.
How can I solve this so I dont have to restart the service everytime before clients can boot?

Greetings and many thanks!

**Comment by alkisg** [comment]
Hi, reboot your server, run the following commands, and paste the output:

```shell
grep -r bind-interfaces /etc/dnsmasq*
systemctl status dnsmasq
```

**Reply by costinpc** [reply]
Hi alkisg,
Here is the output: 
/etc/dnsmasq.conf:#bind-interfaces
/etc/dnsmasq.d/network-manager:bind-interfaces

dnsmasq.service - dnsmasq - A lightweight DHCP and caching DNS server
   Loaded: loaded (/lib/systemd/system/dnsmasq.service; enabled; vendor preset: enabled)
  Drop-In: /run/systemd/generator/dnsmasq.service.d
           └─50-dnsmasq-$named.conf, 50-insserv.conf-$named.conf
   Active: active (running) since Vi 2021-06-11 17:15:52 EEST; 8min ago
  Process: 801 ExecStartPost=/etc/init.d/dnsmasq systemd-start-resolvconf (code=exited, status=0/
  Process: 763 ExecStart=/etc/init.d/dnsmasq systemd-exec (code=exited, status=0/SUCCESS)
  Process: 653 ExecStartPre=/usr/sbin/dnsmasq --test (code=exited, status=0/SUCCESS)
 Main PID: 800 (dnsmasq)
   CGroup: /system.slice/dnsmasq.service
           └─800 /usr/sbin/dnsmasq -x /var/run/dnsmasq/dnsmasq.pid -u dnsmasq -r /var/run/dnsmasq

iun 11 17:15:52 test-machine dnsmasq[800]: started, version 2.75 cachesize 150
iun 11 17:15:52 test-machine dnsmasq[800]: compile time options: IPv6 GNU-getopt DBus i18
iun 11 17:15:52 test-machine dnsmasq[800]: DNS service limited to local subnets
iun 11 17:15:52 test-machine dnsmasq-dhcp[800]: DHCP, IP range 10.20.30.10 -- 10.20.30.25
iun 11 17:15:52 test-machine dnsmasq-tftp[800]: TFTP root is /srv/tftp
iun 11 17:15:52 test-machine dnsmasq[800]: no servers found in /var/run/dnsmasq/resolv.co
iun 11 17:15:52 test-machine dnsmasq[800]: read /etc/hosts - 7 addresses
iun 11 17:15:52 test-machine systemd[1]: Started dnsmasq - A lightweight DHCP and caching
iun 11 17:15:54 test-machine dnsmasq[800]: reading /var/run/dnsmasq/resolv.conf
iun 11 17:15:54 test-machine dnsmasq[800]: using nameserver 192.168.170.2#53

Regards,

**Comment by costinpc** [comment]
I'm reinstalling Lubuntu 16.04 LTS and wanna give it a go again. Can someone tell me the order of the commands that need to be run?
When I first installed ltsp I ran this commands:

apt install --install-recommends ltsp ltsp-binaries dnsmasq nfs-kernel-server openssh-server squashfs-tools ethtool net-tools epoptes
gpasswd -a administrator epoptes

*Another method is to have a dual NIC LTSP server, where one NIC is connected to the normal network where the Internet is, and the other NIC is connected to a separate switch with just the LTSP clients. For this method to work automatically, assign a static IP of 192.168.67.1 to the internal NIC using Network Manager or whatever else your distribution has, and run:

ltsp dnsmasq --proxy-dhcp=0
ltsp image /
ltsp ipxe
ltsp nfs
ltsp initrd

I'm I doing it wrong?
The first time as I said everything was working but after a restart of dnsmasq.service every time I started server.

**Comment by alkisg** [comment, marked as answer]
Sorry I forgot to answer this one. Just delete this file:

```shell
sudo rm -f /etc/dnsmasq.d/network-manager
sudo systemctl restart network-manager
```

There's a bug in dnsmasq in which bind-interfaces isn't working properly, with the result that you're seeing. It's not related to LTSP.

**Reply by costinpc** [reply]
Thanks a lot! It works if I delete the file you said. No longer time out errors.
But now I encounter another issue, after command ltsp dnsmasq --proxy-dhcp=0 I receive the following error and I don't longer have access to internet on server, on the client side I have...
![err](https://user-images.githubusercontent.com/85743946/121896122-9f9aaf00-cd29-11eb-9ce1-4e7ad03ea63c.JPG)
Found out on the server I have only 127.0.0.1 as DNS and that's the reason why no Internet and if I add Google DNS for example after restart it will disappear. Can you instruct me how to add other DNS permanently?

**Comment by costinpc** [comment]
I managed to set DNS permanently. Everything is ok now. Thank you very much for your help!
PS* Is there a better solution for my scenario?  "One server (i7-2600, 8Gb RAM, 2 NIC-s) and 20 clients (Pentium Dual Cores ~2Ghz, 1 GB RAM). Clients need only browsing."

**Comment by alkisg** [comment]
Instead of the expired Ubuntu 16.04, I would use Debian 11 Bullseye 32 bit for the server, in chrootless mode (ltsp image /),
with a tiny desktop environment like openbox, so that the session is just a window manager and the browser.

Although the Internet in 2021 needs at least 2 GB RAM, so I would try to upgrade the RAM first.

**Comment by costinpc** [comment]
I will look into your suggested solution and try to upgrade RAM.
Thank you very much @alkisg for your intel!
